ipamapi/contract.go

```diff
@@ -42,59 +51,22 @@ type AddressSpace string
 // Common key is a addressspace
 type Config interface {
```

Comment:
When initially wrote the ipam package, we saw situations where the provisioning of subnets and the request/release of addresses is not done by the same entity, so I ended up providing two interfaces, so that libnetwork can pass each interface's provider to the proper entity.